### PR TITLE
Pytest

### PR DESCRIPTION
**Pull request by hellysmile**

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+version: '2'
+
+services:
+  cassandra:
+    ports:
+      - 9042:9042
```
Comment (webknjaz) [inline docker-compose.yml]: I normally tend to not add extra indentation to sequences. This helps to deal with nested things better.

File: tests/conftest.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    loop = asyncio.new_event_loop()
+    loop.set_debug(bool(os.environ.get('PYTHONASYNCIODEBUG')))
+
+    yield loop
+
+    loop.call_soon(loop.stop)
+    loop.run_forever()
+    loop.close()
+
+    gc.collect()
```
Comment (webknjaz) [inline tests/conftest.py]: Double this line for PyPy

File: tox.ini [hunk 3]
```diff
@@ -1,20 +1,20 @@
...
 deps =
-    cassandra-driver
     flake8
     isort
 commands =
-    flake8 --show-source aiocassandra.py setup.py tests.py
-    isort --check-only aiocassandra.py setup.py tests.py
+    flake8 --show-source aiocassandra.py setup.py
+    isort --check-only aiocassandra.py setup.py
+    flake8 --show-source tests
```
Comment (webknjaz) [inline tox.ini]: Why not append it to the previous command?

File: tox.ini [hunk 4]
```diff
@@ -1,20 +1,20 @@
...
+    isort --check-only -rc tests --diff
 
-    # envpython to mute "WARNING:test command found but not installed in testenv"
-    {envpython} tests.py
+    {envpython} setup.py pytest
```
Comment (webknjaz) [inline tox.ini]: In fact, using setup.py didn't make it to become a standard protocol for running tests. Turns out that the only best practice nowadays is to just invoke pytest.

File: setup.py [hunk 5]
```diff
@@ -29,6 +32,8 @@ def read(*parts):
     description='Simple threaded cassandra wrapper for asyncio',
     long_description=read('README.rst'),
     install_requires=['cassandra-driver'],
+    setup_requires=['pytest-runner'] if needs_pytest else [],
+    tests_require=['pytest', 'pytest-asyncio', 'pytest-cov'],
```
Comment (webknjaz) [inline setup.py]: I think, you shouldn't really bother adding this.

File: .travis.yml [hunk 6]
```diff
@@ -31,6 +30,9 @@ before_script:
 - tox -e wait-for-cassandra
 script:
 - tox -e python
+after_success:
+  - pip install codecov
```
Comment (webknjaz) [inline .travis.yml]: Please dedent this + reset this step in deploy job.

File: tox.ini (hunk 3 above)
Comment (hellysmile) [reply tox.ini]: isort for folders requires `-rc` , as well split lint code and lint tests in different parts

File: tests/conftest.py (hunk 2 above)
Comment (hellysmile) [reply tests/conftest.py]: good catch!

**Comment by hellysmile** [comment]
@webknjaz sorry for fast merge

Fixes is here https://github.com/aio-libs/aiocassandra/pull/13

File: README.rst [hunk 7]
```diff
@@ -3,12 +3,15 @@ aiocassandra
 
 :info: Simple threaded cassandra wrapper for asyncio
 
-.. image:: https://img.shields.io/travis/aio-libs/aiocassandra.svg
+.. image:: https://travis-ci.org/aio-libs/aiocassandra.svg?branch=master
```
Comment (webknjaz) [inline README.rst]: Why not?

**Pull request merged**